### PR TITLE
Add multiplication to div_mod test.

### PR DESCRIPTION
**Pull request by dsharletg**
This PR adds multiplication to the div_mod test. Hexagon has several special case implementations for multiplication, which were previously only incidentally tested.

I tested this branch with the targets host-hvx_64, host-hvx_128, host-cuda, and host-opencl.

### CODE REVIEW

File: test/correctness/mul_div_mod.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <algorithm>
 
 using namespace Halide;
+using namespace Halide::Internal;
```
Comment (zvookin) [inline test/correctness/mul_div_mod.cpp]: Is it possible to tighten this down to specific symbols without adding too many? I don't have a strict objection to doing this, but narrowing use of Internal both prevents breakages due to coincidental name conflicts and signals to others that this is not a technique to copy.

File: test/correctness/mul_div_mod.cpp [hunk 2]
```diff
@@ -112,9 +114,9 @@ bool is_negative_one(unsigned char val) {
 }
 
 
-template<typename T,typename BIG,int bits>
+template<typename T,typename BIG>
```
Comment (zvookin) [inline test/correctness/mul_div_mod.cpp]: Space after commas on these.

**Comment by zvookin** [comment]
LGTM.

**Comment by dsharletg** [comment]
Sorry to make a change after you looked, I noticed that gpu_vectorize_div_mod.cpp tests a subset of this, so I removed it.

**Pull request merged**